### PR TITLE
feat: [P4PU-162] transaction detail api adaptation

### PR DESCRIPTION
**Pull request by AnisLucidi01**
#### List of Changes
Adapted new API for transaction details. 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/routes/Transaction.tsx [hunk 1]
```diff
@@ -1,25 +1,22 @@
 import React from 'react';
 // import TransactionDetail from '../components/Transactions/TransactionDetail';
```
Comment (stratoivandiluccio) [inline src/routes/Transaction.tsx:2]: remove this unuseful commented line

File: src/utils/converters.test.ts [hunk 2]
```diff
@@ -10,11 +10,11 @@ describe('toEuro function', () => {
 
     expect(utils.converters.toEuro(501)).toEqual('5,01\xa0€');
 
-    expect(utils.converters.toEuro(0)).toEqual('0,00\xa0€');
+    expect(utils.converters.toEuro(0)).toEqual('-');
```
[the remark this answers is not in the document]
Comment (AnisLucidi01) [reply src/utils/converters.test.ts:13]: With Roberto we decided to convert 0€ to "-", as per task P4PU-256

[the remark this answers is not in the document]
Comment (stratoivandiluccio) [reply src/utils/converters.test.ts:13]: Sorry, but can't work in this way. At least we can return '-' when the function receive undefined, not 0.

**Review by mmoio — APPROVED** [review]

File: src/routes/Transaction.tsx [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
-  // const { data, isError } = utils.loaders.getTransactionDetails(id as string);
-  // const transactionDetailData = data && utils.converters.prepareTransactionDetailData(data);
-
+  const { data, isError } = utils.loaders.getTransactionDetails(id as string);
+  const transactionDetailData = data && utils.converters.prepareTransactionDetailData(data);
   return (
     <>
       <QueryLoader
         loaderComponent={<TransactionDetailsSkeleton />}
         fallback={isError && <p>Ops! Something went wrong, please try again</p>}
```
Comment (stratoivandiluccio) [inline src/routes/Transaction.tsx:16]: This component (QueryLoader) doesn't accept a fallback property anymore

**Review by stratoivandiluccio — CHANGES_REQUESTED** [review]
Please check the QueryLoader in src/routes/Transaction

**Review by stratoivandiluccio — APPROVED** [review]

**Pull request merged**